soc/arm/nuvoton_npcx/common/pinctrl_soc.h

```diff
@@ -98,6 +134,14 @@ typedef struct npcx_pinctrl pinctrl_soc_pin_t;
 	COND_CODE_1(DT_PROP(node_id, drive_open_drain),				\
 		(NPCX_DRIVE_TYPE_OPEN_DRAIN), (NPCX_DRIVE_TYPE_PUSH_PULL))
 
+#define  Z_PINCTRL_NPCX_PSL_IN_MODE(node_id)				\
+	COND_CODE_0(DT_ENUM_IDX(node_id, psl_in_mode),			\
+		(NPCX_PSL_IN_MODE_LEVEL), (NPCX_PSL_IN_MODE_EDGE))
```

Comment:
why not use `DT_ENUM_IDX` directly?

Reply:
Just in case the order of prop. `psl-in-mode` and enum `npcx_psl_in_mode` doesn't match. But using `DT_ENUM_IDX(node_id, psl_in_mode)` directly is better if we have more than two selections in the future.

Reply:
Done.